src/libpmem/pmem.c

```diff
@@ -591,6 +609,41 @@ pmem_map(int fd)
 }
 
 /*
+ * memory_align  -- (internal) 64-byte memory alignment
+ */
+static inline int
+memory_align(void *dest, const void *src, int direction)
+{
+	int i;
+	size_t cnt;
+
+	if ((cnt = ((uint64_t)dest & ALIGN_MASK)) == 0)
+		return cnt;
+
+	if (direction == UP) {
+		cnt = FLUSH_ALIGN - cnt;
+		uint8_t *d8 = (uint8_t *)dest;
+		const uint8_t *s8 = (uint8_t *)src;
+		for (i = 0; i < cnt; i++) {
+			*d8 = *s8;
+			d8++;
+			s8++;
+		}
+		pmem_flush(dest, cnt);
+	} else {
+		uint8_t	*d8 = (uint8_t *)dest;
+		const uint8_t *s8 = (uint8_t *)src;
+		for (i = 0; i < cnt; i++) {
+			d8--;
+			s8--;
+			*d8 = *s8;
+		}
+		pmem_flush(dest, cnt);
+	}
+	return cnt;
+}
+
+/*
```

Comment:
So this copy_until_aligned() function still seems unnecessary to me.  You have 8 lines that get called from the one place that calls this function with direction == UP, and you have 8 lines that get called from the one place that calls this function with direction == DOWN.  Just move those two clumps of lines to the one caller and this function goes away, unless I'm totally missing something.